### PR TITLE
Port swift specific compiler-rt code to Windows

### PR DESCRIPTION
**Pull request by hughbe**
https://github.com/apple/swift-compiler-rt/pull/5

We need to cherry pick this to get compiler-rt building on apple/swift-4.0-branch and apple/stable

### CODE REVIEW

File: lib/profile/InstrProfilingUtil.c [hunk 1]
```diff
@@ -225,6 +226,7 @@ COMPILER_RT_VISIBILITY const char *lprofFindLastDirSeparator(const char *Path) {
   return Sep;
 }
 
+<<<<<<< HEAD
```
Comment (jrose-apple) [inline lib/profile/InstrProfilingUtil.c]: ^

Comment (hughbe) [reply lib/profile/InstrProfilingUtil.c]: Oops... thanks

**Comment by hughbe** [comment]
~@swift-ci please test~

Tested locally, and has been pushed ah well

**Comment by jrose-apple** [comment]
@vedantk, can you sanity-check this?

**Comment by vedantk** [comment]
Lgtm!

**Comment by hughbe** [comment]
Can someone merge this? I only have the "Close Pull Request" option - no option to "Merge Pull Request!"

**Comment by jrose-apple** [comment]
Merged, sorry.

**Pull request merged**